kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -719,74 +701,191 @@ impl<'a, 'b> CrateAnalyzer<'a, 'b> {
         Ok(())
     }
 
+    /// Emit the Kythe edges for cross references for imports in this crate
+    pub fn emit_import_xrefs(&mut self) -> Result<(), KytheError> {
+        // We must clone to avoid double borrowing "self"
+        let analysis = self.krate.analysis.clone();
+
+        let template_vname = self.krate_vname.clone();
+        let krate_signature = template_vname.get_signature();
+
+        for reference in &analysis.imports {
+            // If there is no reference id, we can't emit a cross reference
+            if reference.ref_id.is_none() {
+                continue;
+            }
+
+            // Currently this is only known to support `Use` ImportKinds
+            if reference.kind != rls_data::ImportKind::Use {
+                continue;
+            }
+
+            let mut reference_vname = template_vname.clone();
+            let span = &reference.span;
+
+            // Get the CrateId for the referenced crate
+            let ref_id = reference.ref_id.unwrap();
+            let krate_id_option = self.krate_ids.get(&ref_id.krate);
+            if krate_id_option.is_none() {
+                // This is a little bit of chicken and egg here. We could try
+                // and find the file vname first before the crate id, but then
+                // we wouldn't have a definition anchor to emit the diagnostic
+                // node to because we wouldn't have the crate id yet. The chance
+                // of a crate id not being in our HashMap is very low, so if this
+                // fails we just emit a diagnostic node to the first source file
+                // of the crate we are indexing
+                let file_vname = self.file_vnames.values().next().unwrap();
+                self.emitter.emit_diagnostic(
+                    file_vname,
+                    "Cross reference could not be generated",
+                    Some(&format!("Failed to generate cross reference for \"{:?}\" because the referenced crate could not be found", ref_id)),
+                    None
+                )?;
+                continue;
+            }
+            let krate_id = krate_id_option.unwrap();
+
+            // Create VName for target of reference
+            let target_vname = self.create_def_vname(krate_id, ref_id.index);
+
+            // Create VName for the reference node
+            let file_name = span.file_name.to_str().unwrap();
+            let file_vname = self.file_vnames.get(file_name);
+            if file_vname.is_none() {
+                self.emitter.emit_diagnostic(
+                    &target_vname,
```

Comment:
Diagnostics should be pointing from a file or anchor node; isn't target_vname the semantic node? (does this question apply elsewhere?) Maybe these variables just need better names to make that clear, i.e. rename `target_node` to `anchor_node`?

Reply:
Renamed the variable and made sure to convert to an anchor vname before emitting the diagnostic